hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -478,6 +480,10 @@ private void doBucketProbing() throws IOException {
             .getInt(S3A_BUCKET_PROBE, S3A_BUCKET_PROBE_DEFAULT);
     Preconditions.checkArgument(bucketProbe >= 0,
             "Value of " + S3A_BUCKET_PROBE + " should be >= 0");
+    String endPoint = getConf().getTrimmed(ENDPOINT, "");
```

Comment:
lets skip this if mode == 0 as that way DNS lookups don't cause init delays.

Reply:
Done